master/buildbot/changes/github.py

```diff
@@ -0,0 +1,259 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import time
+from datetime import datetime
+
+from twisted.internet import defer
+from twisted.internet import reactor
+
+from buildbot.changes import base
+from buildbot.util import ascii2unicode
+from buildbot.util import datetime2epoch
+from buildbot.util import epoch2datetime
+from buildbot.util import httpclientservice
+from buildbot.util.logger import Logger
+from buildbot.util.state import StateMixin
+
+log = Logger()
+
+HOSTED_BASE_URL = "https://api.github.com"
+
+
+class GitHubPullrequestPoller(base.ReconfigurablePollingChangeSource,
+                              StateMixin):
+    compare_attrs = ("owner", "repo", "token", "branches", "pollInterval",
+                     "useTimestamps", "category", "project", "pollAtLaunch",
+                     "name")
+    db_class_name = 'GitHubPullrequestPoller'
+
+    def __init__(self, owner, repo, **kwargs):
+        name = kwargs.get("name")
+        if not name:
+            kwargs["name"] = "GitHubPullrequestPoller:" + owner + "/" + repo
+        super(GitHubPullrequestPoller, self).__init__(owner, repo, **kwargs)
+
+    def checkConfig(self,
+                    owner,
+                    repo,
+                    branches=None,
+                    pollInterval=10 * 60,
+                    useTimestamps=True,
+                    category=None,
+                    baseURL=None,
+                    project='',
+                    pullrequest_filter=True,
+                    token=None,
+                    encoding='utf-8',
+                    pollAtLaunch=False,
+                    **kwargs):
+        base.ReconfigurablePollingChangeSource.checkConfig(
+            self,
+            name=self.name,
+            pollAtLaunch=pollAtLaunch,
+            pollInterval=pollInterval)
```

Comment:
you forgot to pass kwargs (so that we can add feature to PollingChangeSource later and don't need to modify subclasses_